### PR TITLE
Exposed methods that return instances of vendors and purposes

### PR DESCRIPTION
**Pull request by felipe-saez**
This PR contains changes required to expose the following methods and corresponding integration tests:
```
getDisabledPurposes
getDisabledVendors
getEnabledPurposes
getEnabledVendors
getPurpose
getRequiredPurposes
getRequiredVendors
getVendor
```

In order to expose these methods new custom types had to be created: `Vendor` and `Purpose`

### CODE REVIEW

File: lib/entities/vendor.dart [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        flexiblePurposeIds = json["flexiblePurposeIds"].cast<String>(),
+        specialPurposeIds = json["specialPurposeIds"].cast<String>(),
+        specialFeatureIds = json["specialFeatureIds"].cast<String>(),
+        cookieMaxAgeSeconds = json["cookieMaxAgeSeconds"],
+        usesNonCookieAccess = json["usesNonCookieAccess"],
+        deviceStorageDisclosureUrl = json["deviceStorageDisclosureUrl"];
+
+  @override
+  int compareTo(Vendor other) => (id ?? "").compareTo(other.id ?? "");
+}
```
Comment (nicolas-chaix-didomi) [inline lib/entities/vendor.dart:71]: missing blank line at the end of file

File: lib/entities/purpose.dart [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  Purpose.fromJson(dynamic json)
+      : id = json["id"],
+        iabId = json["iabId"],
+        name = json["name"],
+        description = json["description"],
+        descriptionLegal = json["descriptionLegal"];
+
+  @override
+  int compareTo(Purpose other) => (id ?? "").compareTo(other.id ?? "");
+}
```
Comment (nicolas-chaix-didomi) [inline lib/entities/purpose.dart:25]: missing blank line at the end of file

File: lib/entities/entities_helper.dart [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+import 'package:didomi_sdk/entities/vendor.dart';
+
+// Utility class used to do computations related to vendors and purposes.
+class EntitiesHelper {
+  // Convert a list of dictionaries into a list of purposes.
+  static List<Purpose> toPurposes(List<dynamic> rawList) => rawList.map((e) => Purpose.fromJson(e)).toList();
+
+  // Convert a list of dictionaries into a list of vendors.
+  static List<Vendor> toVendors(List<dynamic> rawList) => rawList.map((e) => Vendor.fromJson(e)).toList();
+}
```
Comment (nicolas-chaix-didomi) [inline lib/entities/entities_helper.dart:11]: missing blank line at the end of file

File: example/lib/widgets/get_vendor.dart [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+  Future<String> callDidomiPlugin() async {
+    final Vendor result = await DidomiSdk.getVendor("google");
+
+    if (result == null) {
+      return "Vendor does not exist.";
+    } else {
+      return "Vendor: ${result.name}.";
+    }
+  }
+}
```
Comment (nicolas-chaix-didomi) [inline example/lib/widgets/get_vendor.dart:32]: missing blank line at the end of file

File: example/lib/widgets/get_required_vendors.dart [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+    final String printable = result.map((vendor) => vendor.name).join(", ");
+    print(printable);
+
+    if (result.isEmpty) {
+      return "Required Vendor list is empty.";
+    } else {
+      return "Required Vendors: $printable.";
+    }
+  }
+}
```
Comment (nicolas-chaix-didomi) [inline example/lib/widgets/get_required_vendors.dart:36]: missing blank line at the end of file

File: example/lib/widgets/get_required_purposes.dart [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+    final String printable = result.map((purpose) => purpose.name).join(", ");
+    print(printable);
+
+    if (result.isEmpty) {
+      return "Required Purpose list is empty.";
+    } else {
+      return "Required Purposes: $printable.";
+    }
+  }
+}
```
Comment (nicolas-chaix-didomi) [inline example/lib/widgets/get_required_purposes.dart:36]: missing blank line at the end of file

File: example/lib/widgets/get_purpose.dart [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+  Future<String> callDidomiPlugin() async {
+    final Purpose result = await DidomiSdk.getPurpose("cookies");
+
+    if (result == null) {
+      return "Purpose does not exist.";
+    } else {
+      return "Purpose: ${result.name}.";
+    }
+  }
+}
```
Comment (nicolas-chaix-didomi) [inline example/lib/widgets/get_purpose.dart:32]: missing blank line at the end of file

File: example/lib/widgets/get_enabled_vendors.dart [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+    final String printable = result.map((vendor) => vendor.name).join(", ");
+    print(printable);
+
+    if (result.isEmpty) {
+      return "Enabled Vendor list is empty.";
+    } else {
+      return "Enabled Vendors: $printable.";
+    }
+  }
+}
```
Comment (nicolas-chaix-didomi) [inline example/lib/widgets/get_enabled_vendors.dart:36]: missing blank line at the end of file

File: example/lib/widgets/get_enabled_purposes.dart [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+    final String printable = result.map((purpose) => purpose.name).join(", ");
+    print(printable);
+
+    if (result.isEmpty) {
+      return "Enabled Purpose list is empty.";
+    } else {
+      return "Enabled Purposes: $printable.";
+    }
+  }
+}
```
Comment (nicolas-chaix-didomi) [inline example/lib/widgets/get_enabled_purposes.dart:36]: missing blank line at the end of file

File: example/lib/widgets/get_disabled_vendors.dart [hunk 10]
```diff
@@ -0,0 +1,36 @@
...
+    final String printable = result.map((vendor) => vendor.name).join(", ");
+    print(printable);
+
+    if (result.isEmpty) {
+      return "Disabled Vendor list is empty.";
+    } else {
+      return "Disabled Vendors: $printable.";
+    }
+  }
+}
```
Comment (nicolas-chaix-didomi) [inline example/lib/widgets/get_disabled_vendors.dart:36]: missing blank line at the end of file

File: example/lib/widgets/get_disabled_purposes.dart [hunk 11]
```diff
@@ -0,0 +1,36 @@
...
+    final String printable = result.map((purpose) => purpose.name).join(", ");
+    print(printable);
+
+    if (result.isEmpty) {
+      return "Disabled Purpose list is empty.";
+    } else {
+      return "Disabled Purposes: $printable.";
+    }
+  }
+}
```
Comment (nicolas-chaix-didomi) [inline example/lib/widgets/get_disabled_purposes.dart:36]: missing blank line at the end of file

File: example/lib/widgets/get_disabled_purposes.dart [hunk 12]
```diff
@@ -0,0 +1,36 @@
...
+/// Widget to call DidomiSdk.getDisabledPurposes
+class GetDisabledPurposes extends StatefulWidget {
+  @override
+  State<StatefulWidget> createState() => _GetDisabledPurposesState();
+}
+
+class _GetDisabledPurposesState
+    extends BaseSampleWidgetState<GetDisabledPurposes> {
+  @override
+  String getButtonName() => 'GetDisabledPurposes';
```
Comment (nicolas-chaix-didomi) [inline example/lib/widgets/get_disabled_purposes.dart:17]: You should use double quotes everywhere (except for import)

**Review by nicolas-chaix-didomi — CHANGES_REQUESTED** [review]

File: lib/entities/purpose.dart [hunk 13]
```diff
@@ -0,0 +1,25 @@
+// Class used to represent a purpose.
+class Purpose implements Comparable<Purpose> {
+  // Purpose ID
+  String? id;
```
Comment (pmerlet-at-didomi) [inline lib/entities/purpose.dart:4]: Could id be not nullable, so we return null or an error if it is not present?

File: lib/entities/vendor.dart [hunk 14]
```diff
@@ -0,0 +1,71 @@
+// Class used to represent a vendor.
+class Vendor implements Comparable<Vendor> {
+  // Vendor ID.
+  String? id;
```
Comment (pmerlet-at-didomi) [inline lib/entities/vendor.dart:4]: Same, if possible it should not be nullable

File: lib/entities/entities_helper.dart (hunk 3 above)
Comment (pmerlet-at-didomi) [inline lib/entities/entities_helper.dart:11]: Missing blank line

File: lib/entities/purpose.dart (hunk 2 above)
Comment (pmerlet-at-didomi) [inline lib/entities/purpose.dart:25]: Missing blank line

File: lib/entities/vendor.dart (hunk 1 above)
Comment (pmerlet-at-didomi) [inline lib/entities/vendor.dart:71]: Missing blank line

File: lib/entities/purpose.dart (hunk 13 above)
Comment (felipe-saez) [reply lib/entities/purpose.dart:4]: we'll keep as nullable for now considering null/nil values can come from native SDKs.

File: lib/entities/vendor.dart (hunk 14 above)
Comment (felipe-saez) [reply lib/entities/vendor.dart:4]: we'll keep this as nullable for now considering null/nil values can come from native SDKs.

File: lib/entities/vendor.dart (hunk 1 above)
Comment (felipe-saez) [reply lib/entities/vendor.dart]: done

Comment (felipe-saez) [reply lib/entities/vendor.dart]: done

**Review by nicolas-chaix-didomi — APPROVED** [review]

**Review by pmerlet-at-didomi — APPROVED** [review]

**Review by nicolas-chaix-didomi — APPROVED** [review]

**Pull request merged**